health/target/target.go

```diff
@@ -0,0 +1,55 @@
+// Package target provides exported structs that you should use during
+// health monitoring initialization and collection.
+package target
+
+import (
+	"errors"
+
+	"github.com/zenoss/zenoss-go-sdk/utils"
+)
+
+// New creates a new Target object
+func New(
+	id string, enableHeartbeat bool, metricIDs, counterIDs, totalCounterIDs []string,
+) (*Target, error) {
+	set := make(map[string]struct{})
+	for _, val := range metricIDs {
+		set[val] = struct{}{}
+	}
+	for _, val := range counterIDs {
+		set[val] = struct{}{}
+	}
+	for _, val := range totalCounterIDs {
+		set[val] = struct{}{}
+	}
+	if len(set) < len(metricIDs)+len(counterIDs)+len(totalCounterIDs) {
+		return nil, errors.New("Some metric or counter ID is not unique")
+	}
+	return &Target{
+		ID:              id,
```

Comment:
Consider adding a type field to allow better health computation flexibility. For example, ZDM (via user config) could give higher health priority to datasources of a particular type vs others when determining it's own health state

Reply:
Covered in ZING-17816